packages/aws-cdk-lib/aws-ec2/lib/instance.ts

```diff
@@ -589,11 +590,23 @@ export class Instance extends Resource implements IInstance {
     this.instancePublicDnsName = this.instance.attrPublicDnsName;
     this.instancePublicIp = this.instance.attrPublicIp;
 
-    if (props.init) {
-      this.applyCloudFormationInit(props.init, props.initOptions);
+    if (props.initOptions?.timeout && props.resourceSignalTimeout) {
```

Comment:
I believe if the feature flag is enabled, so no need for this warning, as this is our recommended behaviour.